x-pack/plugins/monitoring/public/alerts/badge.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment } from 'react';
+import {
+  EuiContextMenu,
+  EuiPopover,
+  EuiBadge,
+  EuiFlexGrid,
+  EuiFlexItem,
+  EuiText,
+} from '@elastic/eui';
+import { CommonAlertStatus, CommonAlertState } from '../../common/types';
+import { AlertSeverity } from '../../common/enums';
+// @ts-ignore
+import { formatDateTimeLocal } from '../../common/formatting';
+import { AlertState } from '../../server/alerts/types';
+import { AlertPanel } from './panel';
+import { Legacy } from '../legacy_shims';
+import { isInSetupMode } from '../lib/setup_mode';
+
+function getDateFromState(states: CommonAlertState[]) {
+  const timestamp = states[0].state.ui.triggeredMS;
+  const tz = Legacy.shims.uiSettings.get('dateFormat:tz');
+  return formatDateTimeLocal(timestamp, false, tz === 'Browser' ? null : tz);
+}
+
+interface Props {
+  alerts: { [alertTypeId: string]: CommonAlertStatus };
+}
+export const AlertsBadge: React.FC<Props> = (props: Props) => {
+  const [showPopover, setShowPopover] = React.useState<AlertSeverity | boolean | null>(null);
+  const inSetupMode = isInSetupMode();
+  const alerts = Object.values(props.alerts).filter(Boolean);
+
+  const badges = [];
+
+  if (inSetupMode) {
+    const button = (
+      <EuiBadge
+        iconType="bell"
+        onClickAriaLabel={`${alerts.length} alert(s)`}
+        onClick={() => setShowPopover(true)}
+      >
+        {alerts.length} alert(s)
+      </EuiBadge>
+    );
+    const panels = [
+      {
+        id: 0,
+        title: `Alerts`,
+        items: alerts.map(({ alert }, index) => {
+          return {
+            name: (
+              <Fragment>
+                <EuiText>{alert.label}</EuiText>
+              </Fragment>
+            ),
+            panel: index + 1,
+          };
+        }),
+      },
+      ...alerts.map((alertStatus, index) => {
+        return {
+          id: index + 1,
+          title: alertStatus.alert.label,
+          width: 400,
+          content: (
+            <div style={{ padding: '1rem' }}>
```

Comment:
Maybe this padded div can be wrapped **inside** the `AlertPanel` component? So we don't have to add it when ever we use it

Reply:
Yea I changed this a bit, thanks!